File: aoc-2023/Day-06/Python/main.py
import math
l = [input().split()[1:] for _ in '..']
g = lambda a,b: a-1-2*int((a-(a*a-4*b)**0.5)/2)
print('Part 1:', math.prod(g(*map(int,i)) for i in zip(*l)))
print('Part 2:', g(*map(int,map(''.join, l))))

File: aoc-2023/Day-06/Ruby/main.rb
# Testing with ChatGPT
def g(a, b)
    a - 1 - 2 * ((a - Math.sqrt(a * a - 4 * b)) / 2).to_i
end

data = []

2.times do
    data << gets.split(/:\s+/)[1].split(/\s+/)
end

p = 1
a = ''
b = ''

data[0].each_with_index do |x, i|
    y = data[1][i]
    p *= g(x.to_i, y.to_i)
    a += x
    b += y
end

puts "Part 1: #{p}"
puts "Part 2: #{g(a.to_i, b.to_i)}"